sdk/security/keyvault/internal/challenge_policy_test.go

```diff
@@ -355,3 +358,64 @@ func TestParseTenant(t *testing.T) {
 	actual = parseTenant(sampleURL)
 	require.Equal(t, expected, actual, "tenant was not properly parsed, got %s, expected %s", actual, expected)
 }
+
+func TestChallengePolicy_ConcurrentRequests(t *testing.T) {
+	concurrentRequestCount := 3
+
+	serverAuthenticateRequests := atomic.Int32{}
+	serverAuthenticatedRequests := atomic.Int32{}
+	var srv *httptest.Server
+	srv = httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Other tests use `mock.Server`, but here I used `httptest.Server` for two reasons:

1. `mock.Server` is documented to not be safe when used from multiple goroutines.
2. Even if `mock.Server` was safe at a surface level (e.g. by adding locks), its interface doesn't allow me to handle out-of-order requests.

`httptest.Server` has its own issues, but it seemed like the least amount of code to get the test working.

Reply:
It's ok to use httptest.Server here.

But unfortunatly, your tests can't use `atomic.Int32{}`, as it was added in Go 1.19. Our team is currently sticking to Go version 1.18. Other than that, your code looks good to me